controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,883 @@
+			return fmt.Errorf(cannotBeNilError, "subscriptionId", config.ClusterName)
+		}
+		if config.Spec.TenantID == "" {
+			return fmt.Errorf(cannotBeNilError, "tenantId", config.ClusterName)
+		}
+		if config.Spec.AzureCredentialSecret == "" {
+			return fmt.Errorf(cannotBeNilError, "azureCredentialSecret", config.ClusterName)
+		}
+		if config.Spec.KubernetesVersion == "" {
+			return fmt.Errorf(cannotBeNilError, "kubernetesVersion", config.ClusterName)
+		}
+	}
+	clientId, clientSecret, err := getSecrets(h.secretsCache, config.Spec.AzureCredentialSecret)
+	if err != nil {
+		return fmt.Errorf("could not get secrets")
+	}
+	if clientId == "" || clientSecret == "" {
+		return fmt.Errorf(cannotBeNilError, "clientId and clientSecret", config.ClusterName)
+	}
+
+	azureAuthorizer, err := newClientAuthorizer(h.secretsCache, config.Spec)
+	if err != nil {
+		return err
+	}
+	serviceClient, err := newServiceClient(azureAuthorizer, config.Spec)
+	if err != nil {
+		return err
+	}
+	listOperators, err := serviceClient.ListOrchestrators(ctx, config.Spec.ResourceLocation, "managedClusters")
+	if err != nil {
+		return err
+	}
+	objOrchestrators := *listOperators.Orchestrators
+	validKubernetesVersion := false
+	for _, obj := range objOrchestrators {
+		if config.Spec.KubernetesVersion == to.String(obj.OrchestratorVersion) {
+			validKubernetesVersion = true
+			break
+		}
+	}
+	if !validKubernetesVersion {
+		return fmt.Errorf("field kubernetesVersion is invalid for non-import [%s] cluster location", config.ClusterName)
+	}
+	if hasAgentPoolProfile(config.Spec) {
+		for _, np := range config.Spec.NodePools {
+			if np.Name == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Name", config.ClusterName)
+			}
+			if np.Count == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Count", config.ClusterName)
+			}
+			if np.MaxPods == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.MaxPods", config.ClusterName)
+			}
+			if np.VMSize == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.VMSize", config.ClusterName)
+			}
+			if np.OsDiskSizeGB == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OsDiskSizeGB", config.ClusterName)
+			}
+			if np.OsDiskType == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OSDiskType", config.ClusterName)
+			}
+			if np.Mode == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Mode", config.ClusterName)
+			}
+			if np.OsType == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OsType", config.ClusterName)
+			}
+		}
+	}
+	if (config.Spec.PublicAccess && config.Spec.PrivateAccess) || (!config.Spec.PublicAccess && !config.Spec.PrivateAccess) {
+		return fmt.Errorf("only one field PublicAccess or PrivateAccess have to be set for [%s] cluster", config.ClusterName)
+	}
+
+	if config.Spec.NetworkPolicy != "" {
+		networkPolicy := config.Spec.NetworkPolicy
+		if networkPolicy != string(containersv211.NetworkPolicyAzure) && networkPolicy != string(containersv211.NetworkPolicyCalico) {
```

Comment:
nit: We _could_ combine this if statements. I do not feel strongly about this due to readability.